promql/parse_test.go

```diff
@@ -192,13 +212,17 @@ var testExpr = []struct {
 		fail:   true,
 		errMsg: "could not parse remaining input \"=~ 1\"...",
 	}, {
-		input:  "-some_metric",
+		input:  `-"string"`,
 		fail:   true,
-		errMsg: "expected type scalar in unary expression, got vector",
+		errMsg: `unary expression only allowed on expressions of type scalar or vector, got "string"`,
 	}, {
-		input:  `-"string"`,
+		input:  `-test[5m]`,
 		fail:   true,
-		errMsg: "expected type scalar in unary expression, got string",
+		errMsg: `unary expression only allowed on expressions of type scalar or vector, got "matrix"`,
+	}, {
+		input:  `*test`,
+		fail:   true,
+		errMsg: "no valid expression found",
```

Comment:
If is a keyword. Keywords are not valid metric names so this fails for another reason - still a regression test from the fuzzing.